test-data/unit/check-expressions.test

```diff
@@ -410,6 +410,53 @@ class B:
 main:3: error: Incompatible types in assignment (expression has type "bool", variable has type "A")
 main:4: error: Incompatible types in assignment (expression has type "bool", variable has type "A")
 
+[case testCmp_python2]
+
+class bool: pass
+
+a, b, c, bo = None, None, None, None # type: (A, B, C, bool)
+bo = a == a
+bo = a != a
+bo = a < b
+bo = a > b
+bo = b <= b
+bo = b <= c
+bo = b >= b
+bo = a >= b
+
+class A:
+    def __cmp__(self, o):
+      # type: ('B') -> bool
+      pass
+class B:
+    def __cmp__(self, o):
+        # type: ('B') -> bool
+        pass
+    def __le__(self, o):
+        # type: ('C') -> bool
+        pass
+class C:
+    pass
+[out]
+main:5: error: Argument 1 to "__cmp__" of "A" has incompatible type "A"; expected "B"
```

Comment:
Can you switch to the `# E: ...` convention? It makes it easier to see what lines are supposed to produce which errors.